examples/gadgets/datasource/mutate/main.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/datasource"
+	gadgetcontext "github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-context"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-service/api"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators"
+	clioperator "github.com/inspektor-gadget/inspektor-gadget/pkg/operators/cli"
+	_ "github.com/inspektor-gadget/inspektor-gadget/pkg/operators/ebpf"
+	ocihandler "github.com/inspektor-gadget/inspektor-gadget/pkg/operators/oci-handler"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators/simple"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/runtime/local"
+)
+
+func do() error {
+	// In this case, we need to set priority to a low value to be sure it's
+	// executed before the cli operator
+	const opPriority = 100
+	myOperator := simple.New("myHandler", simple.OnInit(func(gadgetCtx operators.GadgetContext) error {
+		for _, d := range gadgetCtx.GetDataSources() {
+			// It's possible to add new fields to the datasource
+			// TODO: bool not supported yet
+			isRootF, err := d.AddField("is_root", datasource.WithKind(api.Kind_Uint8))
+			if err != nil {
+				return fmt.Errorf("adding field: %w", err)
+			}
+
+			// It's also possible to change visibility, and other attributes of fields
+			timeStampF := d.GetField("timestamp")
+			timeStampF.SetHidden(true, false)
+
+			// Get accessors to some specific fields
+			uidF := d.GetField("uid")
+
+			d.Subscribe(func(source datasource.DataSource, data datasource.Data) error {
+				uid := uidF.Uint32(data)
+				// if uid is zero, set is_root to true
+				if uid == 0 {
+					// TODO: yes, we know, it has to be improved.
+					isRootF.Set(data, make([]byte, 1))
+					isRootF.PutUint8(data, 1)
```

Comment:
This can already be updated now that we merged https://github.com/inspektor-gadget/inspektor-gadget/pull/2823/files

Reply:
Good one, it was even failing to compile.